fix: fix menu position when scrollbar is not visible (#1404)

david-ding opened the pull request:
**Problem**
When the body has full height and we use `appendTo="body"`, when opening the select dropdown, the dropdown menu is being misplaced.
![image](https://user-images.githubusercontent.com/3939207/93759033-d0dbc800-fc4c-11ea-9be5-68e5dde5552a.png)

This issue can be demonstrated at https://stackblitz.com/edit/ng-select-position-issue

**Cause**
When appending the dropdown menu to another container, we are waiting for the dom element to be inserted before calculating the left offset. Problem with this is after appending the menu, it could have triggered the scrollbar to appear, and our left offset is calculated based on when scrollbar is shown. And after we re-positioned the menu, the scrollbar disappears and we end up with misaligned menu.

**Fix**
Separated updates to the horizontal and vertical offset, so that we calculate the left offset before appending the menu, and calculate the vertical offset after appending as usual. The horizontal position of the dropdown should not be relevant to the dropdown menu content, there seems to be no need to wait for the menu to be inserted into DOM first before calculating it.

NicholasMaestrello commented:
I think this is a good solution. I've already get this problem in a application and needed to resolve it localy.

joaoroyosilva reviewed (commented):
I think this is a good solution.

miki995 commented:
@varnastadeus  can you have a look on it, seems like an important stuff?

david-ding commented:
I have just added a test case for this fix. @varnastadeus / @anjmao if you could take a look at this it would be great. thanks!

varnastadeus commented:
@david-ding can you add a proper example where this problem is addressed? you can add it here `src/demo/app/examples/append-to-example/append-to-example.component.html`

I copied your code from stackblitz, and I don't it fixed. 

![image](https://user-images.githubusercontent.com/17824796/95104856-01d4f600-073f-11eb-9d0a-a7766e4d73af.png)

This is the code I used from your example, except I removed height from `ng-select` as there is no point having it there
```html
<div class="container" style="position: relative; overflow: auto; width: 200px; height: 500px">
    <ng-select [items]="people | async"
               bindLabel="company"
               placeholder="Select item"
               appendTo=".container"
               style="width: 50%; margin-left: auto"
               [closeOnSelect]="false"
               [(ngModel)]="selected2">
    </ng-select>
    <div style="border: 1px solid black; height: 400px"></div>
</div>
```

david-ding commented:
Hi @varnastadeus , my apologies - I was playing around with the stackblitz example, didn't realize I altered the original one..I've now reverted it back

I have now added an example of the issue in the demo. The circumstances for the problem to appear is:
- We have a scrollable container div, and the dropdown panel is appended to it
- Before the dropdown panel is appended, the scrollbar is not shown
- After the dropdown panel is appended, the scrollbar is shown
- After the dropdown is re-positioned below the ng-select component, the scrollbar is again not shown

![image](https://user-images.githubusercontent.com/3939207/95142010-66836580-07be-11eb-8ff4-7caa9d935130.png)
When the ng-select component is left aligned, the width is wrong

![image](https://user-images.githubusercontent.com/3939207/95142086-97639a80-07be-11eb-8b97-db2c807c1b2d.png)
When the ng-select component is not left aligned, both positioning and width are wrong

![image](https://user-images.githubusercontent.com/3939207/95142129-b5c99600-07be-11eb-9ccf-d67e48f91fe0.png)
![image](https://user-images.githubusercontent.com/3939207/95142140-bbbf7700-07be-11eb-8be2-2b56421f5685.png)
After fix applied, the positioning and width are correct

This scenario would be relatively common when we appendTo='body' and the body has full height.

varnastadeus reviewed (approved):
the example you provided may confuse some users, especially when you start selecting multiple items but the calculations are more correct now 👍 solid work 💯

david-ding commented:
Thanks @varnastadeus!

That example does look bad when we start to make selections...I have submitted a new PR to simplify that demo, think that's clearer and more concise.
